test/unit/validate_test.py

```diff
@@ -0,0 +1,175 @@
+        pv.weighted_quantile(data_series, weights_series, quantile)
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile",
+    [
+        # All NaN data
+        ([np.nan, np.nan, np.nan], [1, 1, 1], 0.5),
+        # All zero weights
+        ([1, 2, 3], [0, 0, 0], 0.5),
+        # Empty series
+        ([], [], 0.5),
+    ],
+)
+def test_weighted_quantile_nan_cases(data, weights, quantile):
+    """Test that weighted_quantile returns NaN for degenerate cases."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    result = pv.weighted_quantile(data_series, weights_series, quantile)
+    assert np.isnan(result)
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile,expected",
+    [
+        # Data with inf values - function filters out inf but processes remaining data
+        ([1, np.inf, 3], [1, 1, 1], 0.5, 2.0),
+        ([1, -np.inf, 3], [1, 1, 1], 0.5, 2.0),
+        # NaN weights - function filters out rows with NaN weights
+        ([1, 2, 3], [1, np.nan, 1], 0.5, 2.0),
+        # Mix of NaN and inf - function filters out problematic rows
+        ([1, np.nan, np.inf, 4], [1, 1, 1, 1], 0.5, 2.5),
+    ],
+)
+def test_weighted_quantile_with_invalid_data(data, weights, quantile, expected):
+    """Test that weighted_quantile handles invalid data by filtering it out."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    result = pv.weighted_quantile(data_series, weights_series, quantile)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "column,threshold,n_outliers_allowed,expected_pass",
+    [
+        # Test cases that should PASS
+        ("stable_metric", 0.1, 0, True),  # Very stable values, low threshold
+        ("gradual_growth", 0.2, 0, True),  # Gradual 10% growth per year, 20% threshold
+        ("sudden_jump", 5.0, 0, True),  # Large jump but threshold accommodates it
+        ("volatile_metric", 0.3, 1, True),  # Volatile but allowing 1 outlier
+        ("negative_change", 0.2, 0, True),  # Consistent 10% decline, 20% threshold
+        # Test cases that should FAIL
+        ("sudden_jump", 0.1, 0, False),  # Large jump (5x) exceeds 10% threshold
+        ("volatile_metric", 0.1, 0, False),  # High volatility exceeds 10% threshold
+        ("sudden_jump", 1.0, 0, False),  # Large jump exceeds even 100% threshold
+        ("gradual_growth", 0.05, 0, False),  # 10% growth exceeds 5% threshold
+    ],
+)
+def test_group_mean_continuity_check(
+    column, threshold, n_outliers_allowed, expected_pass
+):
+    """Test the group_mean_continuity_check function with various scenarios.
+
+    Uses a test dataframe with different column patterns:
+    - stable_metric: Values around 100 with minimal variation
+    - gradual_growth: Steady 10% growth per year
+    - sudden_jump: Large 5x jump from 2022 to 2023
+    - volatile_metric: Random fluctuations around 100
+    - negative_change: Consistent 10% decline per year
```

Comment:
I'm not sure of the implementation details here; is the % change computed just at the beginning of the sequence or at every data point? bc if it's the latter then for a consistent decline of 100 in absolute terms, the % change increases from 10% to ~15% by the time it reaches the end. Does that matter?